Set a constant chunk size

dkillick opened the pull request:
Removes cleverness in setting variable (bytes and size-based) chunk sizes in favour of fixing a chunk size of 8 bytes. This works around a problem where performing operations on arrays with differing dtypes causes a broadcasting error because the chunks end up having differing sizes.

Also improves the `_all_slices` tests and documentation because these changes flowed from confirming that the right approach was being followed to make this change.

Closes #161 and fixes #164.

rhattersley commented on biggus/__init__.py:
```diff
@@ -2043,33 +2042,51 @@ def ndarrays(arrays):
+    # MAX_CHUNK_SIZE. nbytes will be updated as we traverse the dimensions of
+    # shape so that it equals the number of bytes that one item in the current
+    # dimension represents.
+    nbytes = 8
```
It doesn't make sense to me to keep the concept of bytes. Really you're moving from counting bytes to counting array elements.

rhattersley replied:
In other words, I'd rename this variable as something like `n_elements` and start with `n_elements = 1`

dkillick commented:
Replaced `nbytes` (number of bytes to chunk on) with the more appropriate concept of `n_elems` (number of elements to chunk on), and have reduced its starting value to 1. 

Have made necessary test changes and added an admittedly complex integration test that captures a previously-failing edge case that has been fixed by the change to `n_elems = 1`.

dkillick commented:
Hmm, though I still can't get the Travis tests working :angry: All the tests pass locally!

pelson commented:
> Hmm, though I still can't get the Travis tests working :angry: All the tests pass locally!

You tested py3k, right?

dkillick commented:
Ah! Much better :grinning:

marqh commented:
Hi @dkillick 

the tests are passing but there are merge conflicts, please may you rebase and push?

thank you

dkillick commented:
Rebased.

Bump? @pelson @rhattersley
